### PR TITLE
General Clean Up

### PR DESCRIPTION
**Pull request by robputt**
This PR provides a general clean up prior to me actually starting to write tests.

- Fixes Flake8 Compliance Issues (auto generated files excluded via noqa comments)
- Removes committed _pycache_ directories
- Updates Github Actions for more strict enforcement of the style guide
- Removes nmcli from the codebase, this appears to be a duplicate of the upstream PyPi package nmcli, instead lets install via requirements when building the container image - https://pypi.org/project/nmcli/
- Add bandit for security audit.

https://github.com/NebraLtd/hm-config/issues/33

### CODE REVIEW

File: config-python/bletools.py [hunk 1]
```diff
@@ -23,38 +21,44 @@
...
 
+
 class BleTools(object):
     @classmethod
     def get_bus(self):
-         bus = dbus.SystemBus()
-
-         return bus
+        bus = dbus.SystemBus()
+        return bus
```
Comment (vpetersson) [inline config-python/bletools.py:38]: `return dbus.SystemBus()`?

File: requirements.txt [hunk 2]
```diff
@@ -4,4 +4,5 @@ colorzero==2.0
 gpiozero==1.6.2
 protobuf==3.15.6
 dbus-python==1.2.16
+nmcli==0.5.0
```
Comment (vpetersson) [inline requirements.txt:7]: @ryanteck what version of `nmcli` is the above code from?

Comment (robputt) [reply requirements.txt:7]: Diff vs upstream nmcli

(.env) robe8437@Roberts-MacBook-Pro nmcli % diff -r nmcli nmcli_nebra     
diff -r nmcli/_system.py nmcli_nebra/_system.py
38c38
<                           check=True, env={'LANG': 'C'})
---
>                           check=True)
Only in nmcli_nebra/data: __pycache__
(.env) robe8437@Roberts-MacBook-Pro nmcli %

**Review by vpetersson — APPROVED** [review]
Good clean-up, @robputt

**Review by ryanteck — CHANGES_REQUESTED** [review]
NMCli was included seperately for that exact code change to fix a bug.

This will break it.

**Comment by vpetersson** [comment]
> NMCli was included seperately for that exact code change to fix a bug.
> 
> This will break it.

Can you elaborate on the bug? is this diff above?

**Comment by ryanteck** [comment]
> > NMCli was included seperately for that exact code change to fix a bug.
> > This will break it.
> 
> Can you elaborate on the bug? is this diff above?

It didn't work, the diff above fixed it and that's why I did it. Sorted

These changes will break the software.

**Comment by robputt** [comment]
We should pass the discussed above as an environment variable in the Docker container...

nmcli wraps Python sub process and as demonstrated below the sub processes inherit the environment variables from the parent...

```
robe8437@Roberts-MacBook-Pro hm-config % export LANG=C
robe8437@Roberts-MacBook-Pro hm-config % env | grep LANG
LANG=C
robe8437@Roberts-MacBook-Pro hm-config % python3
Python 3.9.4 (default, Apr  5 2021, 01:50:46) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from subprocess import CalledProcessError, run
>>> run('env')
TERM_PROGRAM=Apple_Terminal
SHELL=/bin/zsh
TERM=xterm-256color
TMPDIR=/var/folders/yh/mm13v7f127sdz7qz_9blsmlr0000gn/T/
TERM_PROGRAM_VERSION=440
TERM_SESSION_ID=91CDE78A-FE47-4F9C-ADC3-C5590C137809
USER=robe8437
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.dP60FIWkUu/Listeners
PATH=/Users/robe8437/.nvm/versions/node/v12.22.1/bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
__CFBundleIdentifier=com.apple.Terminal
PWD=/Users/robe8437/Workspace/Projects/hm-config
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
SHLVL=1
HOME=/Users/robe8437
LOGNAME=robe8437
OLDPWD=/Users/robe8437/Workspace/Projects/hm-config
NVM_DIR=/Users/robe8437/.nvm
NVM_CD_FLAGS=-q
NVM_BIN=/Users/robe8437/.nvm/versions/node/v12.22.1/bin
NVM_INC=/Users/robe8437/.nvm/versions/node/v12.22.1/include/node
LANG=C
_=/usr/local/bin/python3
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x2
LC_CTYPE=UTF-8
LINES=74
COLUMNS=165
CompletedProcess(args='env', returncode=0)
>>> 
```

I will update the Dockerfile accordingly.

**Review by vpetersson — APPROVED** [review]
LGTM @robputt. Can you also post the Python shell snippet you posted to me before in a comment here for future references?

**Comment by robputt** [comment]
> LGTM @robputt. Can you also post the Python shell snippet you posted to me before in a comment here for future references?

See https://github.com/NebraLtd/hm-config/pull/34#issuecomment-867015939

**Comment by robputt** [comment]
@ryanteck - The changes above to set the environment variable should alleviate the issue without having to duplicate the nmcli package. I validated that the subprocesses also inherit the environment as seen here https://github.com/NebraLtd/hm-config/pull/34#issuecomment-867015939

Best Regards,

Rob

**Comment by ryanteck** [comment]
I'm sorry but I'm not reviewing any new code and accepting PRs. This would be down to @vpetersson to test and merge.

**Comment by vpetersson** [comment]
Given that you still have the test hardware in your possession, nobody else can test it. Besides, as you have discussed with @shawaj, you will help with the hand over.

**Comment by shawaj** [comment]
> I'm sorry but I'm not reviewing any new code and accepting PRs. This would be down to @vpetersson to test and merge.

@ryanteck please can you help with this? We really need to ensure a smooth handover.

**Comment by ryanteck** [comment]
I cannot gaurantee the validty of any new code and I do not feel that I am able to certify any new code which is not my responisiblity now.

This process was requested by @vpetersson so he has assumed responsibility for maintaining this code and therefore should organise the process to test it from now on.

**Comment by shawaj** [comment]
> should organise the process to test it from now on.

And I've requested that you test it on the test devices you have available to you. So would appreciate if you could do that.

We are a team, we work together. You are still part of the team currently.

File: .github/workflows/security-audit.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Code Security Audit
```
Comment (ryanteck) [inline .github/workflows/security-audit.yml:1]: This seems to possibly not even be required? From what I can gather this is primarily targeted towards Django.

I'd prefer changes which aren't required to not be made while I'm still being asked to review them.

**Review by ryanteck — APPROVED** [review]
I believe most of the changes actually seem to be with the base BTLE code / library and the change of switching nmcli back to pip.

To test will be that it's merged in and deployed to testnet.

**Pull request merged**